code/mob/living.dm

```diff
@@ -1344,6 +1344,11 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
 		if(src.chat_text.vis_locs.len)
 			var/atom/movable/AM = src.chat_text.vis_locs[1]
 			AM.vis_contents -= src.chat_text
+
+			//Fixes spooky floating text dup with spooky code
+			if((istype(AM, /obj/machinery) || istype(AM, /obj/artifact)) && src.client)
+				AM.invisibility = 101
+				AM.invisibility = 0
```

Comment:
i would be very surprised if this is different from just setting to 0.

Reply:
Doesn't seem to work if it isn't made fully invisible first, (just tested). The duplication issue goes away when the player gets far enough from the object where they can't see it, but if you make it fully invisible it also resolves that issue.